src/DocumentationBrowserViewExtension/MarkdownHandler.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Dynamo.PackageManager;
+using Markdig;
+using Markdig.Parsers;
+using Markdig.Renderers;
+
+namespace Dynamo.DocumentationBrowser
+{
+    internal class MarkdownHandler
+    {
+        private const string NODE_ANNOTATION_NOT_FOUND = "Dynamo.DocumentationBrowser.Docs.NodeAnnotationNotFound.md";
+        private MarkdownPipeline pipeline;
+
+
+        private static MarkdownHandler instance;
+        public static MarkdownHandler Instance
+        {
+            get
+            {
+                if (instance is null) { instance = new MarkdownHandler(); }
+                return instance;
+            }
+        }
+
+        public MarkdownHandler()
+        {
+            var pipelineBuilder = new MarkdownPipelineBuilder();
+            pipeline = pipelineBuilder
+                .UseAdvancedExtensions()
+                .Build();
+        }
+
+        internal void GetAnnotationFromMd(ref StringWriter writer, string mdFilePath)
+        {
+            if (writer is null)
+                throw new ArgumentNullException(nameof(writer));
+
+            string mdString;
+            if (string.IsNullOrEmpty(mdFilePath))
+                mdString = GetNodeAnnotationNotFoundContent();
+
+            else
+            {
+                var fileInfo = new FileInfo(mdFilePath);
+                while (IsFileLocked(fileInfo))
```

Comment:
issue came up when modifying an open the file file displaying it in dynamo (hot reload). Just realized now that this is probably because the `watcher.Changed` fires twice whenever i make an edit - this i apparently a known "issue" https://stackoverflow.com/questions/1764809/filesystemwatcher-changed-event-is-raised-twice 

There dosent seem to be a good solution to it though. I guess there is a potential for this to be stuck in a infinite loop, but in 99% of the cases it just needs to wait for the first event to finish. 

Could a solution be to set a max amount of times it can check if the file is locked? if it reaches the max it will return `File cannot be read`?